dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/IdentifiableObjectBundleHook.java

```diff
@@ -73,6 +78,58 @@ public void preCreate(IdentifiableObject identifiableObject, ObjectBundle bundle
     handleAttributeValues(identifiableObject, bundle, schema);
     handleSkipSharing(identifiableObject, bundle);
     handleSkipTranslation(identifiableObject, bundle);
+    handleSortOrder(identifiableObject, bundle, schema);
+  }
+
+  /**
+   * This method loops through all sortable List of given object and sets the sortOrder value for
+   * each item in the List if it is not already set.
+   *
+   * @param identifiableObject Object to set sortOrder on
+   * @param bundle {@link ObjectBundle}
+   * @param schema Schema of given object
+   */
+  private void handleSortOrder(
+      IdentifiableObject identifiableObject, ObjectBundle bundle, Schema schema) {
+    findSortableProperty(schema)
+        .forEach(
+            property -> {
+              List<IdentifiableObject> collection =
+                  ListUtils.emptyIfNull(
+                      ReflectionUtils.invokeGetterMethod(
+                          property.getFieldName(), identifiableObject));
+              for (int i = 0; i < collection.size(); i++) {
+                IdentifiableObject item = collection.get(i);
+                IdentifiableObject preheatedItem =
+                    bundle.getPreheat().get(bundle.getPreheatIdentifier(), item);
+                if (preheatedItem == null) {
+                  continue;
+                }
+                SortableObject sortableObject = (SortableObject) preheatedItem;
+                if (sortableObject.getSortOrder() == null) {
+                  sortableObject.setSortOrder(i);
```

Comment:
I wonder if we should make this a bit smarter. Since we only set the order for objects which did not have one, we might not want to set it too `i` but to previous item's value + 1?

Reply:
Thought about this, i think it doesn't matter where those `null` item are put by the backend, User will eventually need to rearrange it so I made the code to be as simple as possible. 
Anyway, updated the code and added the index variable. We can re-evaluate this logic later.